spec/views/item_seasons/show.html.erb_spec.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'item_seasons/show', type: :view do
+  let!(:item_season_produces) { ProduceItem.create(name: 'carambola', category: 'fruit') }
+  let!(:item_season) { ItemSeason.create(produce_item_id: item_season_produces.id, month_index: 1, country_code: 'br') }
+
+  it 'display month name' do
+    assign(:item_season, item_season)
+    assign(:item_season_produces, ProduceItem.joins(:item_seasons)
+    .where(item_seasons: { month_index: item_season.month_index }))
```

Comment:
**nitpick:**  Indentation looks off to me

We could move `ProduceItem.joins(:item_seasons).where(item_seasons: { month_index: item_season.month_index })` to a variable. If you prefer not to, I'd usually go for

```rb
    assign(
      :item_season_produces, 
      ProduceItem.joins(:item_seasons).where(item_seasons: { month_index: item_season.month_index })
    )
```

Reply:
Agreed, but the way you suggested makes rubocop complain about 'too many lines'. What do you think about this? 
```
    assign(:item_season_produces, ProduceItem.joins(:item_seasons)
                                             .where(item_seasons: { month_index: item_season.month_index }))

```

I tried moving it to a variable, but couldn't get it to work

Reply:
TBH I think we should tweak Rubocop's configuration to be less strict in this aspect. Would you mind sharing the error here?

Reply:
```
Offenses:

spec/views/item_seasons/show.html.erb_spec.rb:17:3: C: RSpec/ExampleLength: Example has too many lines. [7/5]
  it 'displays the produces' do ...
```

We had this problem once, and we excluded the file in the rubocop.yml, but maybe you should change Rubocop's configuration regarding this rule.

```
RSpec/ExampleLength:
  Exclude:
    - 'spec/views/produce_items/show.html.erb_spec.rb'
```

Reply:
do you think we should set a new max? 11 would work for our current long tests. Or maybe should we disable this test?

Reply:
I was messing around a bit, and tried using `CountAsOne`, it takes our `produce_items/show.html.erb_spec.rb` from 11 lines to 8, so doesn't solve it, and doesn't do anything for the other two. Still, it could be helpfull. 

Right now all our problems are in the views, we could exclude that folder. But this problem could appear again in the future in another test.

Right now I have something like this:
```
RSpec/ExampleLength:
  CountAsOne:
    - array
    - hash
    - heredoc
  Max: 5
  Exclude:
    - 'spec/views/**/*.rb' 
```

WDYT?

Reply:
I like it :). However `Max: 5` seems very little to me. WDYT about moving it to 10?

Reply:
another solution could be changing the tests, this works:

```
  before do
    assign(:item_season, item_season)
    assign(
      :item_season_produces,
      ProduceItem.joins(:item_seasons).where(item_seasons: { month_index: item_season.month_index })
    )
  end

  it 'display month name' do
    render
    expect(rendered).to match(/Janeiro/)
  end

  it 'displays the produces' do
    render
    expect(rendered).to match(/Carambola/)
  end
```

Reply:
Hm, the structure you showed in the comment before looks better to me (since you are DRYing away the assigns). Let's go with it AND w/ the increased limit.

Reply:
I was reading how in tests sometimes repetition was warranted if it promoted readability. I think here the tests are super simple, and separating the assigns doesn't really make it harder to read. But I was wandering which way was best to write it, and ended leaving it there 😬